src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,181 @@
+        JobStatusType statusType = JobStatusType.PROCESSED;
+
+        for (CourtList courtList : dailyList.getCourtLists()) {
+
+            String courtHouseName = courtList.getCourtHouse().getCourtHouseName();
+            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(
+                    courtHouseName);
+
+            if (foundCourthouse.isPresent()) {
+                List<Sitting> sittings = courtList.getSittings();
+                for (Sitting sitting : sittings) {
+                    List<Hearing> hearings = sitting.getHearings();
+                    for (Hearing dailyListHearing : hearings) {
+
+                        String caseNumber = getCaseNumber(dailyListHearing.getDefendants(), dailyListEntity, dailyListHearing);
+
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName, String.valueOf(sitting.getCourtRoomNumber()),
+                                caseNumber, dailyListHearing.getHearingDetails().getHearingDate()
+                        );
+
+                        CourtCaseEntity courtCase = hearing.getCourtCase();
+                        addJudges(sitting, hearing);
+                        addDefendants(courtCase, dailyListHearing.getDefendants());
+                        addProsecutions(courtCase, dailyListHearing.getProsecution().getAdvocate().getPersonalDetails().getName());
+                        addDefenders(courtCase, dailyListHearing.getDefendants());
+
+                        courtCase.addHearing(hearing);
+
+                        hearingRepository.saveAndFlush(hearing);
+                        caseRepository.saveAndFlush(courtCase);
+                    }
+                }
+            } else {
+                statusType = JobStatusType.PARTIALLY_PROCESSED;
+                log.error("Unregistered courthouse " + courtHouseName + " daily list entry has not been processed");
+            }
+        }
+        dailyListEntity.setStatus(statusType.name());
+    }
+
+    private String getCaseNumber(List<Defendant> defendants, DailyListEntity dailyListEntity, Hearing hearing) {
+        // CPP don't provide case id, use URN
+        if (String.valueOf(SourceType.CPP).equalsIgnoreCase(dailyListEntity.getSource())) {
+            if (defendants.isEmpty()) {
+                return hearing.getCaseNumber();
+            } else {
+                String urn = defendants.get(0).getUrn();
+                if (StringUtils.isEmpty(urn)) {
+                    log.error("Hearing not added - HearingInfo does not contain a URN value");
+                } else {
+                    return urn;
+                }
+            }
+        }
+        return hearing.getCaseNumber();
+
+    }
+
+
+    private void addProsecutions(CourtCaseEntity courtCase, CitizenName prosecutionName) {
```

Comment:
as its only adding one, should this be `addProsecutor`?

Reply:
ACCEPT